recipes/openblas/all/conanfile.py

```diff
@@ -137,7 +157,9 @@ def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "OpenBLAS")
         self.cpp_info.set_property("cmake_target_name", "OpenBLAS::OpenBLAS")
         self.cpp_info.set_property("pkg_config_name", "openblas")
-        cmake_component_name = "pthread" if self.options.use_thread else "serial" # TODO: ow to model this in CMakeDeps?
+        cmake_component_name = "pthread" if self.options.use_thread else "serial"  # TODO: how to model this in CMakeDeps?
+        self.cpp_info.components["openblas_component"].set_property(
+            "cmake_target_name", "OpenBLAS::" + cmake_component_name)  # 'pthread' causes issues without namespace
```

Comment:
I'm a bit confused about what this is doing - do you know why this is necessary? It might be a bit strange to add `OpenBLAS::pthread` and `OpenBLAS::serial` as targets - I can't imagine this is what anybody would expect.

Having said that, this seems to be a reasonable deviation from what upstream CMake suggests, which would be more akin to [[1](https://cmake.org/cmake/help/latest/module/FindBLAS.html)][[2](https://cmake.org/cmake/help/latest/module/FindLAPACK.html)]:

```cmake
set(BLA_VENDOR OpenBLAS)
find_package(BLAS)
find_package(LAPACK)

target_link_libraries(mylib BLAS::BLAS LAPACK::LAPACK)
```

So, what this package is doing is unexpected anyway, and I don't have a clear idea how we would even model the above in conan - maybe this is the best way to handle it

Reply:
The reason for adding the targets as I have suggested is because they are defined in the upstream [OpenBLAS config targets](https://github.com/OpenMathLib/OpenBLAS/blob/9f815cf1bf16b4e64d4aee681b33558fc090b62a/cmake/OpenBLASConfig.cmake.in#L19). Or, that is, they do not have the `OpenBLAS::` namespace. `pthread` as target name causes issues without this prefix on Linux. I do not know why it works for the native OpenBLAS target script, but not for the conan generated one.

It could be useful to add [cmake target aliases](https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#properties) besides the standard OpenBLAS target names, e.g. `BLAS::BLAS` and `LAPACK::LAPACK` as you suggest. We should at the very least retain the `OpenBLAS::OpenBLAS` target, which an OpenBLAS would expect.

Reply:
Fair enough. We could probably add aliases for `BLAS::BLAS` and `LAPACK::LAPACK` in this recipe as well (I think this would be suitable in the case where a consumer links against either of these), but I think matching the interface above by setting `BLA_VENDOR` is probably beyond the capabilities of `CMakeDeps` generator at the moment, so don't worry about this too much. I think we can defer these changes for a future PR. I know there were issues with complexity in the previous PR, so lets keep this one simple and targeted.

I hadn't realised that OpenBLAS exported its own config file, so thanks for directing me to that.